src/misc/support.cpp

```diff
@@ -342,6 +342,35 @@ const std::vector<std_fs::path>& get_resource_parent_paths()
 	return paths;
 }
 
+// Searches mostly the same parent paths as get_resource_parent_paths()
+std::vector<std_fs::path> get_plugin_paths()
+{
+	// Intentionally not using the static cache as this function only gets called once.
+	std::vector<std_fs::path> paths = {};
+
+	// Current working directory
+	maybe_add_path(std_fs::path(PluginsDir), paths);
+
+	maybe_add_path(get_executable_path() / PluginsDir, paths);
+	maybe_add_path(get_executable_path() / ".." / PluginsDir, paths);
+
+	maybe_add_path(std_fs::path(CUSTOM_DATADIR) / DOSBOX_PROJECT_NAME / PluginsDir, paths);
+
+	#if !defined(WIN32) && !defined(MACOSX)
+	maybe_add_path(get_xdg_data_home() / DOSBOX_PROJECT_NAME / PluginsDir, paths);
+
+	for (const auto& data_dir : get_xdg_data_dirs()) {
+		maybe_add_path(data_dir / DOSBOX_PROJECT_NAME / PluginsDir, paths);
+	}
+	#endif
+
+	maybe_add_path(get_executable_path() / "../share" / DOSBOX_PROJECT_NAME / PluginsDir, paths);
```

Comment:
Should this belong to the `#if !defined(WIN32) && !defined(MACOSX)` block as it's a Linux-special thing?

Reply:
It does, but that's only used for "system level" tools, Unixy command line tools, but not by desktop applications with a GUI. Generally, desktop apps live in a different world, and they're roughly similar to FlatPaks.

So no, `/usr/share` lookups are not needed on macOS, we're a desktop app with a GUI, not a low-level system service or a CLI-only too.

<img width="683" height="530" alt="image" src="https://github.com/user-attachments/assets/ecae1c32-e36f-4f56-860d-e6147b9c7802" />

Reply:
Good to know. I added that line to the `#if` block and for resources as well. I think it really only makes sense on non-hipster Unix's like Linux and BSD.